pkg/controller/clusterpool/collections.go

```diff
@@ -584,6 +599,252 @@ func (cds *cdCollection) Delete(c client.Client, cdName string) error {
 	return nil
 }
 
+type cdcCollection struct {
+	// Unclaimed by any cluster pool CD any not broken
+	unassigned []*hivev1.ClusterDeploymentCustomization
+	// Listed in the cluster pool inventory but not found
+	missing []string
+	// Used by some cluster deployment
+	reserved []*hivev1.ClusterDeploymentCustomization
+	// Last Cluster Deployment failed on provision
+	cloud []*hivev1.ClusterDeploymentCustomization
+	// Failed to apply patches for this cluster pool
+	syntax []*hivev1.ClusterDeploymentCustomization
+	// All CDCs in this pool
```

Comment:
That is, all CDCs that were named in this pool's inventory *and* actually exist on the server, right? (I don't know how to phrase this better.)

Reply:
changed to:

>  ByCDCName are all the CDCs listed in the pool inventory, the CR exists and are mapped by name